### PR TITLE
Add ability to provision generic services which are provider agnostic with ability for VMs to span multiple providers based on selected location and location tags

### PR DESCRIPTION
**Pull request by itewk**

### CODE REVIEW

File: Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/create_provision_requests.rb [hunk 1]
```diff
@@ -0,0 +1,215 @@
+# Uses `$evm.execute('create_provision_request', ..)` to create new VMs using the given information.
+# 
+# @param parsed_dialog_options    String (YAML) Parsed dialog options
+# @param parsed_dialog_tags       String (YAML) Parsed dialog tags
+# @param custom_vm_fields         Hash          Additional custom VM fields. See $evm.execute('create_provision_request')
+# @param custom_additional_values Hash          Additional custom additional (ws) fields. See $evm.execute('create_provision_request')
```
Comment (curt-matthews) [inline Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/create_provision_requests.rb]: Additional custom additional?

Comment (itewk) [reply Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/create_provision_requests.rb]: yes, they are additional "additional fields" where "additional fields" is a specific thing as part of an `$evm.execute('create_provision_request',...)`

File: Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/create_provision_requests.rb [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+def yaml_data(task, option)
+  task.get_option(option).nil? ? nil : YAML.load(task.get_option(option))
+end
+
+# Creates provision requests based on the givin parameters.
+#
+# @param task                     ServiceTemplateProvisionTask Task used to create the service that will own the VMs created by these provision request(s)
+# @param requester                User                         User requesting the new VMs
+# @param number_of_vms            Integer                      Number of VMs to provision
+# @param template_fields            Hash                         Hash describing the template to use.
```
Comment (curt-matthews) [inline Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/create_provision_requests.rb]: spacing... if you care

Comment (itewk) [reply Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/create_provision_requests.rb]: fixed in latest PR

File: Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/check_provisioned.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+$evm.log("info", "Listing Root Object Attributes:")
+$evm.root.attributes.sort.each { |k, v| $evm.log("info", "\t#{k}: #{v}") }
+$evm.log("info", "===========================================")
+
+# Get current provisioning status
+task = $evm.root['service_template_provision_task']
+task_status = task['status']
+result = task.statemachine_task_status
+
+$evm.log('info', "Service ProvisionCheck returned <#{result}> for state <#{task.state}> and status <#{task_status}>")
```
Comment (henrygrover) [inline Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/check_provisioned.rb]: ProvisionCheck-> Provision check

Comment (itewk) [reply Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/check_provisioned.rb]: fixed.

File: Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/set_custom_provision_task_options.rb [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+#
+# @return Hash of custom custom additional_values (ws_values) to set on any vm provisioning requests created for the service being created
+def get_custom_additional_values(dialog_options)
+  custom_additional_values = {
+  }
+  
+  return custom_additional_values
+end
+
+# IMPLIMENTORS: do not modify
```
Comment (henrygrover) [inline Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/set_custom_provision_task_options.rb]: implimentor -> implementor throughout the file

Comment (itewk) [reply Automate/RedHatConsulting_Utilities/Service/Provisioning/StateMachines/Methods.class/__methods__/set_custom_provision_task_options.rb]: fixed

**Comment by itewk** [comment]
@henrygrover fixed your comment, can you approve again?

**Comment by itewk** [comment]
re-based

**Comment by itewk** [comment]
@curt-matthews and @henrygrover ping ping

**Comment by henrygrover** [comment]
Sorry didn't see this! Approving.

**Pull request merged**